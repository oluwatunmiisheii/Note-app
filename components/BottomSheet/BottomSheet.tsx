import {
  View,
  StyleSheet,
  Dimensions,
  ScrollView,
  TouchableWithoutFeedback,
} from "react-native";

const bottomSheetHeight = Dimensions.get("window").height * 0.7;
const deviceWidth = Dimensions.get("window").width;

interface BottomSheetProps {
  isOpen: boolean;
  setIsOpen: (value: boolean) => void;
  children: React.ReactNode;
}

export const BottomSheet = ({
  isOpen,
  setIsOpen,
  children,
}: BottomSheetProps) => {
  return isOpen ? (
    <>
      <TouchableWithoutFeedback onPress={() => setIsOpen(false)}>
        <View
          style={{
            ...StyleSheet.absoluteFillObject,
            backgroundColor: "rgba(0,0,0,0.8)",
          }}
        />
      </TouchableWithoutFeedback>
      <View style={styles.bottomSheet}>
        <View style={styles.expandableLine} />
        <View style={styles.bottomSheetContent}>
          <ScrollView showsVerticalScrollIndicator={false}>
            <View>{children}</View>
          </ScrollView>
        </View>
      </View>
    </>
  ) : null;
};

const styles = StyleSheet.create({
  bottomSheet: {
    backgroundColor: "white",
    position: "absolute",
    right: 0,
    left: 0,
    bottom: 0,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    zIndex: 2,
    height: "auto",
    width: deviceWidth,
    maxHeight: bottomSheetHeight,
  },
  bottomSheetContent: {
    paddingVertical: 30,
    width: "100%",
    flex: 1,
    position: "relative",
  },
  bottomSheetTitle: {
    fontSize: 22,
    fontWeight: "700",
    marginBottom: 30,
  },
  expandableLine: {
    width: 50,
    height: 5,
    backgroundColor: "grey",
    borderRadius: 10,
    alignSelf: "center",
    marginTop: 10,
  },
});
