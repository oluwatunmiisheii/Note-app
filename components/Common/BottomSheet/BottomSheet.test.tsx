import { render, screen } from "@testing-library/react-native";
import { BottomSheet } from "./BottomSheet";
import { Text } from "react-native";

const setIsOpen = jest.fn();
const children = <Text>Test</Text>;

const defaultProps: React.ComponentProps<typeof BottomSheet> = {
  isOpen: true,
  children,
  setIsOpen: setIsOpen,
};

const setup = (props: Partial<typeof defaultProps> = defaultProps) => {
  const utils = render(<BottomSheet {...{ ...defaultProps, ...props }} />);
  return { props, ...utils };
};

describe(`<${BottomSheet.name}>`, () => {
  test("renders children", async () => {
    setup();
    expect(screen.getByText("Test")).toBeOnTheScreen();
  });

  test("should not render if isOpen is false", async () => {
    setup({ isOpen: false });
    expect(screen.queryByText("Test")).not.toBeOnTheScreen();
  });
});
