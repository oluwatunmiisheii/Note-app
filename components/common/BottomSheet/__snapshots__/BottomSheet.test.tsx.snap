// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomSheet> renders correctly 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.8)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />,
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "bottom": 0,
        "height": "auto",
        "left": 0,
        "maxHeight": 933.8,
        "position": "absolute",
        "right": 0,
        "width": 750,
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "grey",
          "borderRadius": 10,
          "height": 5,
          "marginTop": 10,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 30,
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View>
          <View>
            <Text>
              Test
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>,
]
`;
